core/src/test/java/projectzero/UMLClassManagerTest.java

```diff
@@ -63,4 +73,39 @@ public void testUMLClassSetName(){
         Assertions.assertEquals("project", umlClass.getName());
     }
 
+    @Test
+    public void testLoadReturnsTrueOnSuccess(@Mock UMLClassYamlMapper umlClassYamlMapper) throws IOException {
+        when(umlClassYamlMapper.read(anyString())).thenReturn(anyList());
+
+        UMLClassManager umlClassManager = new UMLClassManager(umlClassYamlMapper);
+
+        Assertions.assertTrue(umlClassManager.load("test.yaml"));
+    }
+
+    @Test
+    public void testLoadReturnsFalseOnFailure(@Mock UMLClassYamlMapper umlClassYamlMapper) throws IOException {
+        doThrow(IOException.class).when(umlClassYamlMapper).read(anyString());
+
+        UMLClassManager umlClassManager = new UMLClassManager(umlClassYamlMapper);
+
+        Assertions.assertFalse(umlClassManager.load("test.yaml"));
+    }
+
+    @Test
+    public void testSaveReturnsTrueOnSuccess(@Mock UMLClassYamlMapper umlClassYamlMapper) throws IOException {
+        doNothing().when(umlClassYamlMapper).write(anyString(), anyList());
+
+        UMLClassManager umlClassManager = new UMLClassManager(umlClassYamlMapper);
+
+        Assertions.assertTrue(umlClassManager.save("test.yaml"));
+    }
+
+    @Test
+    public void testSaveReturnsFalseOnFailure(@Mock UMLClassYamlMapper umlClassYamlMapper) throws IOException {
+        doThrow(IOException.class).when(umlClassYamlMapper).write(anyString(), anyList());
+
+        UMLClassManager umlClassManager = new UMLClassManager(umlClassYamlMapper);
+
+        Assertions.assertFalse(umlClassManager.save("test.yaml"));
+    }
 }
```

Comment:
Would love to run through this testing with someone to understand it more. I can only assume its testing what we want it testing.

Reply:
Yes, it's mocking the UmlClassMapper read/write methods so that real files are not being written to / read from. The mock is then injected (dependency injection) into the UMLClassManager through the constructor.